client/softlayer_client_test.go

```diff
@@ -70,7 +70,7 @@ var _ = Describe("SoftLayerClient", func() {
 			if urlErr, ok := err.(*url.Error); ok {
 				err = urlErr.Err
 			}
-			Expect(err).To(Equal(errDialFailed))
+			Expect(err.Error()).To(ContainSubstring("dial failed"))
```

Comment:
Why are we only looking for `"dail failed"` as a test for this change? I would expect a new test that shows that a error with message that includes API key is masked, so a text like: `"12345abcd@softlayer.com"` changes to `"************@api.softlayer.com"`.

Please add such a test and I will merge. Thanks.